crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,437 @@
+    /// The space around the inside of the UI element
+    pub padding: UiRect<Val>,
+    /// The space around the outside of the UI element that can be colored to create a visible border
+    pub border: UiRect<Val>,
+}
+
+impl Spacing {
+    ///```rust
+    /// assert_eq!(Spacing::DEFAULT, Spacing::default())
+    ///```
+    pub const DEFAULT: Spacing = Spacing {
+        margin: UiRect::DEFAULT,
+        padding: UiRect::DEFAULT,
+        border: UiRect::DEFAULT,
+    };
+
+    /// Adds an automatic margin on all sides.
+    ///
+    /// When the element is explicitly sized, this will (hopefully) cause the element
+    /// to be vertically and horizontally centered relative to its parent.
+    pub const AUTO_MARGIN: Spacing = Spacing::margin(UiRect {
+        left: Val::Auto,
+        right: Val::Auto,
+        bottom: Val::Auto,
+        top: Val::Auto,
+    });
+
+    /// Sets only the margin
+    pub const fn margin(rect: UiRect<Val>) -> Spacing {
+        Spacing {
+            margin: rect,
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets the margin on all sides to `val`
+    pub const fn margin_all(val: Val) -> Spacing {
+        Spacing {
+            margin: UiRect {
+                left: val,
+                right: val,
+                bottom: val,
+                top: val,
+            },
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets only the padding
+    pub const fn padding(rect: UiRect<Val>) -> Spacing {
+        Spacing {
+            padding: rect,
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets the padding on all sides to `val`
+    pub const fn padding_all(val: Val) -> Spacing {
+        Spacing {
+            padding: UiRect {
+                left: val,
+                right: val,
+                bottom: val,
+                top: val,
+            },
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets only the padding
+    pub const fn border(rect: UiRect<Val>) -> Spacing {
+        Spacing {
+            border: rect,
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets the border on all sides to `val`
+    pub const fn border_all(val: Val) -> Spacing {
+        Spacing {
+            border: UiRect {
+                left: val,
+                right: val,
+                bottom: val,
+                top: val,
+            },
+            ..Self::DEFAULT
+        }
+    }
+}
+
+/// Defines the text direction
+///
+/// For example English is written LTR (left-to-right) while Arabic is written RTL (right-to-left).
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum TextDirection {
+    /// Inherit from parent node
+    #[default]
+    Inherit,
+    /// Text is written left to right
+    LeftToRight,
+    /// Text is written right to left
+    RightToLeft,
+}
+
+/// Whether to show or hide overflowing items
```

Comment:
In the future you should be able to set this per axis and scroll should be an option. But that's outside the scope of this PR, just a note for later